src/object_properties.cpp

```diff
@@ -171,4 +175,12 @@ void ObjectProperties::deSerialize(std::istream &is)
 	try {
 		damage_texture_modifier = deSerializeString(is);
 	} catch (SerializationError &e) {}
+	if (is.eof()) {
+		return;
+	}
+	nametag_viewthrough_walls = readU8(is);
+	if (is.eof()) {
+		return;
+	}
+	nametag_viewthrough_liquids = readU8(is);
```

Comment:
You can read a single `u8` using a bitflag. You'll only need a single check then.